r/tests/testthat/test-arrow-pacakge.R

```diff
@@ -0,0 +1,21 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+test_that("on_rosetta() does not warn", {
+  # There is no warning
+  expect_warning(on_rosetta(), NA)
```

Comment:
I wasn't actually able to find a way to test that the console output from stderr doesn't show here. I tried both `expect_output(..., NA)` and `expect_silent(...)` as well as capturing the output with `capture_output(...)` and `capture.output(...)` but none of those seemed to actually capture the console warning I'm seeing. This is _probably_ fine. I've tested locally and the output is supressed on my system.